### PR TITLE
[tests-only] [full-ci] remove web smoke tests from nightly and full ci

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
This PR removes `Web-Tests-ocis-smoke-ocis-storage-` pipelines from `nightly` and PR title with `full-ci`.

## Related Issue

- https://github.com/owncloud/ocis/issues/8794

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -1086,6 +1086,9 @@ def uiTests(ctx):
             if (not debugPartsEnabled or (debugPartsEnabled and runPart in skipExceptParts)):
                 pipelines.append(uiTestPipeline(ctx, "", runPart, numberOfParts))
 
+    if (ctx.build.event == "cron" or "full-ci" not in ctx.build.title.lower()):
+        return pipelines
+
     # For ordinary PRs, always run the "minimal" UI test pipeline
     # That has its own expected-failures file, and we always want to know that it is correct,
     if (ctx.build.event != "tag"):
```
Comment (saw-jan) [inline .drone.star:1094]:
```suggestion
    # For ordinary PRs, always run the "minimal" UI test pipeline
    # That has its own expected-failures file, and we always want to know that it is correct,
    if (ctx.build.event != "tag" and ctx.build.event != "cron" and "full-ci" not in ctx.build.title.lower()):
```

**Review by saw-jan — APPROVED** [review]

File: .drone.star [hunk 2]
```diff
@@ -1088,7 +1088,7 @@ def uiTests(ctx):
 
     # For ordinary PRs, always run the "minimal" UI test pipeline
     # That has its own expected-failures file, and we always want to know that it is correct,
-    if (ctx.build.event != "tag"):
+    if (ctx.build.event != "tag" and ctx.build.event != "cron" and "full-ci" not in ctx.build.title.lower()):
```
Comment (phil-davis) [inline .drone.star:1091]: `!= "cron"` will stop these from being run in the nightly "cron" CI job - that is a requirement here, good.

`"full-ci" not in ctx.build.title.lower()` means that if someone puts `full-ci` in their PR title, then these smoke test pipelines will not run. It is possible that the code in a PR has changed something that makes a smoke test fail, but we won't find that out. I think that will be surprising, when one day a PR has green CI, gets merged, and then the smoke tests in the master branch are failing.

Comment (saw-jan) [reply .drone.star:1091]: Currently, only smoke webUI tests run in normal PRs (without `full-ci`). If the PR has `full-ci` then `all + smoke` webUI tests pipelines will run which is redundant.
So, we wanted to not run the tests redundantly. Only run all webUI pipeline or smoke webUI pipeline
Same goes for nightly

Comment (phil-davis) [reply .drone.star:1091]: Do whatever is the requirement.

**Review by ScharfViktor — APPROVED** [review]

**Comment by micbar** [comment]
Just for my understanding: what is the reasoning behind that change?

**Comment by saw-jan** [comment]
> Just for my understanding: what is the reasoning behind that change?

with full-ci and cron ci builds, we are running both all-webUI pipelines and smoke-webUI pipelines. If we are running all-webUI pipelines then running smoke-webUI pipelines is redundant, because all test scenarios will run with all-webUI pipelines including smoke scenarios.
So we just wanted to remove the redundancy

**Review by SagarGi — APPROVED** [review]
LGTM :+1:

File: .drone.star (hunk 2 above)
Comment (phil-davis) [reply .drone.star:1091]: OK, I think that I understand now. The "smoke tests" are maybe just test scenarios in within the ordinary test suites that are tagged as smoke tests. In that case, it is redundant to run them twice.

**Pull request merged**